### ISSUE TITLE
Django's `TextChoices` is being treated as a tuple rather than a string in the latest version

### ISSUE THREAD

**Issue by tylerlaprade** [microsoft/pyright#10606]
**Describe the bug**
TextChoices is now being treated as a tuple, even though the revealed type from the `django-types` stub is `str`

**Code or Screenshots**

![Image](https://github.com/user-attachments/assets/125e558d-cf30-45b2-83c4-56e472f33a99)

![Image](https://github.com/user-attachments/assets/c883f2ea-fb1c-454a-8f0e-6661f879b1a1)

```
# requirements.txt
Django == 5.1.6
django-types == 0.20.0
```

```python
from django.db.models import TextChoices


class FooBar(TextChoices):
    FOO = 'F', _('Foo')
    BAR = 'B', _('Bar')

if 'F' == FooBar.FOO:  # reportUnnecessaryComparison: Condition will always evaluate to False since the types "Literal['F']" and "tuple[Literal['F'], Unknown]" have no overlap
    print('yes')
```

**VS Code extension or command-line**
Pylance with the Pyright version explicitly set to `1.1.402`

**Comment by last-partizan** [comment]
@erictraut please take a look here.

If this is an error in `django-types`, let me know I'll fix types there. But probably it's not, we don't have anything fancy there for this case.

**Comment by tylerlaprade** [comment]
@last-partizan I dug through the implementation a bit and this is a weird case. It looks like the equality operator is overridden to compare to both a tuple, and also just the first value of the tuple, so it can actually act as either one.

**Comment by erictraut** [comment]
This is because django is using a custom metaclass `ChoicesType` to override the already-magical `EnumMeta` to provide even-more-magical-and-nonstandard behaviors. A type checker like pyright isn't going to understand that the tuples supplied in the class definition are processed in the `__new__` method of the and only the first tuple element will become the value of the enum member.

So, I don't think I'd categorize this as a bug in `django-types`. It's just a case of a library straying too far from standard behavior for it to work well with standards-based static type checkers.

As a workaround, you could use the following class definition, which should work with static type checkers:

```python
class FooBar(TextChoices):
    FOO = "F"  # , "Foo"
    BAR = "B"  # , "Bar"
```

**Comment by last-partizan** [comment]
Using such class definitions defeats the purpose of TextChoices.

Main goal of TextChoices is to couple enum values with display labels.

Code sample in [pyright playground](https://pyright-play.net/?code=MQAgDgngTglg5gCwC4C4QGcmwMZIFAwC2YA9lEiAKYB2AroXgGZQmEjYkA2nluMJ1dADoAhgCNsIIqXIgAypQCOtGtkpMWbJBDAxqcKcTIUAgtQgAaEABUdlAGoioVqJQBulEZwD62sOrxvAFE6QgBZSkIxSihrEABeGztHKAAKACJg0IiomOt0gEo8PGxOEXR0EABhBBIYNXQIpBFUmnohEPomkQKUPBABkGoRQkp0NE4YTABtTCgAXX7BsujOcZBJmbnFwZAAE0pGEG9vDmpmvXQT1PRKTkYrGCRItDMIApAAWgA%2BEDESLhoITApYDAACYBY-nIEFB%2B0OIDcXhU6Bud0YaD8lGmWS6kWisXmHx%2B8iUKmoahxnXC%2BLy8yBIN2EKhMW0cIOR2wtXqYzR90xdip2VphOJvwUylU2KQtDAPCFeNysSs23pIGBQmKpXKlRqdQarVCHVCVlGzW1FXiep5jUozV67IRJzm11u9zFIDeDM1TMhJGhbN2HI24jufMYHrm3rhzP9rNhQYRSM4KnDHq96pBJTKFRslAAHkhrQa5lZi2MHb6WTDHUdk6m3RGvr8o5nNdmdSAAGIAgBCTlS1gLRe5DUrgy7AHlJwkQAByLtzqwLgFz4quDxeXx2VI9kj9qBCKeTop4UBBABKF7QVQEeye-GoIAA7jBuCAvM%2BRBBKu5kSJnhAJASG7LxbgwPQ1CAhBKCAux0DPEB0gAGSeGIvGmBc53mdIP2oPYkJlOVsVQ54oAwrD5isUj0M4TC92wnCQAQEQPCGECSA8KAyjAAZUlcGQkAAVWoaheDGdAnAgW9iCcKYBCKbBykoABGWcsISRI9wPI9p0QgBJJA50qEgAGtvysOAAT2IQQF7WgKGfBAICkI5IXcdhWDAOT0AEEBGBEN90AAfhKZSVO8FS0DmWdtKcXTJ2KUAADkQJiFgoGYmJKFCnVWXnRdNO7Pt4uPJKvk%2BT4tRzSo9zkJA9kNdp6qgalxwGY91MXcrUqoKAMqy1wrALfxcEoPZLOszUlNuAAmLq5yKuqGoSoA). (exact types as used in django-types)

```python
# pyright: strict
import enum
from collections.abc import Sequence
from typing import Any, TypeVar, reveal_type

_EnumMemberT = TypeVar("_EnumMemberT")

class ChoicesMeta(enum.EnumMeta):
    names: list[str]
    labels: list[str]
    def __contains__(self, item: Any) -> bool: ...
    @property
    def values(self: type[_EnumMemberT]) -> Sequence[_EnumMemberT]: ...
    @property
    def choices(self: type[_EnumMemberT]) -> Sequence[tuple[_EnumMemberT, str]]: ...

class Choices(enum.Enum, metaclass=ChoicesMeta):
    def __str__(self) -> Any: ...
    @property
    def label(self) -> str: ...
    @property
    def value(self) -> Any: ...

class TextChoices(str, Choices):
    @property
    def value(self) -> str: ...

class FooBar(TextChoices):
    FOO = 'F', 'Foo'

reveal_type(FooBar.FOO)

# ERR: Condition will always evaluate to False since the types
# "Literal['F']" and "tuple[Literal['F'], Literal['Foo']]" have no overlap  (reportUnnecessaryComparison)
case1 = 'F' == FooBar.FOO
# It's okay, good. But why if prev comparison fails?
case1_1: str = FooBar.FOO

# No error here?
assert 'F' == FooBar.FOO

# ---

class FooStd(enum.StrEnum):
    FOO = 'F'

# No error here, expected, good.
case2 = 'F' == FooStd.FOO
```

note that "TextChoices" is subtype of str, and I expect it would be compatible/comparable with Literal str.

I understand the argument about "library straying too far", but it was working before, and using `FooBar.FOO.value` instead of `FooBar.FOO`, or dropping labels from definition feels wrong.

**Comment by rob-levy-minimum** [comment]
Even if the equality condition is impossible to support, there's a basic type check which used to work and now doesn't.

Elements of a `TextChoices` class used to declare their type as being instances of that class:

```python
class Animal(models.TextChoices):
    DOG = ("dog", "Dog")
    CAT = ("cat", "Cat")

def greet_animal(animal: Animal):
    ...

greet_animal(Animal.DOG)  # this used to type-check ok

# But now we get:
#      "tuple[Literal['dog'], Literal['Dog']]" is not assignable to "Animal" [reportArgumentType]
```

Should I open a separate issue or is that related to this one?
